### PR TITLE
Initial draft of docs

### PR DESCRIPTION
**Pull request by shannonxtreme**
Added Design Requirements topic and experimented with folder structure and TOC. 
Made changes to the Design Requirements topic and added comments to source file in `Overview/design-requirements.rst`
Added requirements for the site introduction page.

### CODE REVIEW

File: Docs/source/Overview/design-requirements.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Design Requirements
+====================
+Read the regulations at `American Solar Challenge <https://www.americansolarchallenge.org/regulations/2020-american-solar-challenge-regulations/>`_ before creating a BPS. 
+
+Design requirements vary based on the type of batteries you use. For example, lead-acid packs require less protection than lithium-ion cells. BeVolt’s battery pack uses Samsung INR18650-30Q lithium-ion cells. The BPS is required to monitor the current, voltage, and temperature of the battery pack.
+
+.. 
+    Packs vs cells - do lead-acid batteries only come as single units or can you get lead-acid cells? Can a battery pack contain one of either li-ion cells or lead-acid cells? If so,
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-requirements.rst:8]: Lithium-ion and lead-acid batteries both have cells. Multiple cells are combined to create a pack. I see the issue here where we are comparing lithium-ion **cells** to lead-acid **packs.** The idea we are trying to get across is that a lithium-ion battery is less stable than a lead-acid battery, so you need more protection for lithium-ion than lead-acid.

File: Docs/source/Overview/design-requirements.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Design Requirements
+====================
+Read the regulations at `American Solar Challenge <https://www.americansolarchallenge.org/regulations/2020-american-solar-challenge-regulations/>`_ before creating a BPS. 
+
+Design requirements vary based on the type of batteries you use. For example, lead-acid packs require less protection than lithium-ion cells. BeVolt’s battery pack uses Samsung INR18650-30Q lithium-ion cells. The BPS is required to monitor the current, voltage, and temperature of the battery pack.
+
+.. 
+    Packs vs cells - do lead-acid batteries only come as single units or can you get lead-acid cells? Can a battery pack contain one of either li-ion cells or lead-acid cells? If so, 
+    we need to be specific. 
+    Link to cell datasheet would be good.
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-requirements.rst:10]: Cell datasheet is here: https://www.18650batterystore.com/v/files/samsung_30q_data_sheet.pdf

I also have a pdf on our team Google Drive, if that would be better.

File: Docs/source/Overview/design-requirements.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    Link to cell datasheet would be good. 
+
+The BPS should have the following components and features:
+
+    * Minimum coverage on the module level (groups of cells in parallel)
+    * Fuse rated for 200% of the typical current draw of the motors
+    * Trip system that disconnects the battery from the car when:
+        
+        * the voltage of a module exceeds the safe range
+        * the temperature exceeds the maximum specified in the datasheets
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-requirements.rst:19]: We also disconnect the battery from the car when the current through the battery exceeds the safe bounds.

File: Docs/source/Overview/design-requirements.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        .. note::
+            The temperature limit varies depending on whether the battery is charging or discharging.
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-requirements.rst:21]: The current limit also depends on if the battery is charging or discharging.

File: Docs/source/introduction.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Battery Protection System Design 
+================================
+Provide an introduction to the system. The introduction should cover the following topics in brief: 
+
+    1. What is a BPS?
```
Comment (ClarkPoon) [inline Docs/source/introduction.rst:5]: I am guessing this is a list of questions that my team needs to answer for you?

**Review by ClarkPoon — COMMENTED** [review]
This looks nice. I added some comments to fill in some missing information.

**Comment by shannonxtreme** [comment]
@ClarkPoon I added a new section and implemented your comments. Also added links and a Glossary to link terms to. 

For the image showing the entire system, highly recommend we make the board/component names consistent with the documentation. For example, the Current board is called the Amperes board in the image. If you have a source file for the image I can tweak it!

File: Docs/source/Overview/design-overview.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  The minion boards all use `isolated SPI <https://www.analog.com/en/products/interface-isolation/isolation/spisolator.html>`_. 
+.. 
+  There's no flow between the isolated SPI sentence and the following sentence. 
+
+The Current board sends two analog signals to the master board. Each minion board can evaluate 12 modules and 16 temperature sensors.
+
+A Display board is fixed on the dashboard. The BPS Display board is separate from the dashboard to keep it independent from all other systems while debugging.
+
+.. 
+  Does this mean there are two Display boards? Or is the one in your car not on the dashboard?
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-overview.rst:53]: @SijWoo what did you have in mind here?

File: Docs/source/Overview/design-overview.rst [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+Board functions
+===============
+
+The minion boards will be daisy-chained to minimize wire clutter from the master board. 
+The Current (Amps) minion board uses :term:`Hall effect sensors<Hall effect sensor>` to measure the current in the high-voltage line. 
+
+.. note::
+  The minion boards all use `isolated SPI <https://www.analog.com/en/products/interface-isolation/isolation/spisolator.html>`_. 
+.. 
+  There's no flow between the isolated SPI sentence and the following sentence.
```
Comment (ClarkPoon) [inline Docs/source/Overview/design-overview.rst:46]: This information is outdated, and the minion boards and amperes board (formerly known as Current board) all use isolated SPI, so we can get rid of "The Current board sends two analog signals to the master board"

**Comment by ClarkPoon** [comment]
> @ClarkPoon I added a new section and implemented your comments. Also added links and a Glossary to link terms to.
> 
> For the image showing the entire system, highly recommend we make the board/component names consistent with the documentation. For example, the Current board is called the Amperes board in the image. If you have a source file for the image I can tweak it!

Thanks, it looks nice.

I agree 100%. We recently changed the name of the Current board to the Amperes board because we felt that it was confusing on when "current board" meant "board that measures electric current" or "most recent version of board." Unfortunately, we have not had a chance to change the name in all of our documentation yet.

File: Docs/source/Overview/design-overview.rst (hunk 6 above)
Comment (SijWoo) [reply Docs/source/Overview/design-overview.rst:53]: Sorry for the confusion, there are two displays in the car: dashboard display and BPS display. One is on the dashboard that displays speed and cruise control modes; the other display is the BPS display board which displays the **battery percentage, current flow in amps, total voltage, average temperature, and an indicator if the BPS faulted**. The BPS Display is connected to the main CAN Bus. The BPS Display is powered from the BPS, so whenever a trip happens and the battery pack is disconnected from the electrical system, the BPS display is still powered. The BPS display will be in the cockpit but probably not fixed on the dashboard. The BPS display will be mounted on an arm that can be adjusted.

**Review by ClarkPoon — APPROVED** [review]
I recognize that there is still room for improvement here, but I want to merge this into master, so the rest of the BPS team can use this as a guide for how to add to the documentation. What do you think @shannonxtreme @SijWoo @chaseblock ?

**Pull request merged**